channelDiff.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"log"
+	"fmt"
+	"github.com/nlopes/slack"
+	"github.com/google/go-cmp/cmp"
+	"gopkg.in/mgo.v2"
+)
+
+type ChannelAndID struct {
+	Channel Channel
+	ID string
+}
+
+func announceChannelDifferences(coll *mgo.Collection, api *slack.Client, dest string) error {
+	log.Print("Announce channel differences since last run:")
+
+
+	// Get current channels (and their IDs)
+	rawCurrent, err := api.GetChannels(true) // true ignores archived channels
+	if err != nil {
+		return err
+	}
+	current := make([]ChannelAndID, 0, 50)
```

Comment:
Setting length to 0 and capacity to 50 allows us to avoid the performance hit from the first 50 calls to `append`. This can be useful in some cases but it's unnecessary here. Set the length to `len(rawCurrent)` and don't use `append`. You can do a `for i, v := range rawCurrent` and use the `i` to write to `current[i]` directly. No copying, no allocation, and no overhead of calling `append`.

Reply:
Whoops! I feel dumb for missing this one.

Reply:
No need to feel dumb!

Reply:
:D